extras/bindings/fortran/include/precice/SolverInterfaceFortran.hpp

```diff
@@ -160,444 +160,446 @@ PRECICE_API void precicef_requires_writing_checkpoint_(
 PRECICE_API void precicef_has_mesh_(
     const char *meshName,
     int *       hasMesh,
-    int         lengthMeshName);
-
-/**
- * Fortran syntax:
- * precicef_get_mesh_id(
- *   CHARACTER meshName(*),
- *   INTEGER   meshID )
- *
- * IN:  meshName
- * OUT: meshID
- *
- * @copydoc precice::SolverInterface::getMeshID()
- *
- */
-PRECICE_API void precicef_get_mesh_id_(
-    const char *meshName,
-    int *       meshID,
-    int         lengthMeshName);
+    int         meshNameLengthName);
 
 /**
  * Fortran syntax:
  * precicef_has_data(
+ *   CHARACTER meshName(*),
  *   CHARACTER dataName(*),
- *   INTEGER   meshID,
  *   INTEGER   hasData)
  *
- * IN:  dataName
- * IN:  meshID
+ * IN:  mesh, data, meshNameLength, dataNameLength
  * OUT: hasData(1:true, 0:false)
  *
  * @copydoc precice::SolverInterface::hasData()
  *
  */
 PRECICE_API void precicef_has_data_(
+    const char *meshName,
     const char *dataName,
-    const int * meshID,
     int *       hasData,
-    int         lengthDataName);
-
-/**
- * The given name (dataName) has to be one of the names specified in the
- * configuration file. The data id obtained can be used to read and write
- * data to and from the coupling mesh.
- *
- * Fortran syntax:
- * precicef_get_data_id(
- *   CHARACTER dataName(*),
- *   INTEGER   meshID,
- *   INTEGER   dataID,
- *   INTEGER   lengthDataName)
- *
- * IN:  dataName
- * IN:  meshID
- * OUT: dataID
- *
- * @copydoc precice::SolverInterface::getDataID()
- *
- */
-PRECICE_API void precicef_get_data_id_(
-    const char *dataName,
-    const int * meshID,
-    int *       dataID,
-    int         lengthDataName);
+    int         meshNameLength,
+    int         dataNameLength);
 
 /**
  * Fortran syntax:
- * precicef_has_data(
- *   INTEGER   meshID
+ * precicef_precicef_requires_mesh_connectivity_for_(
+ *   CHARACTER meshName(*),
  *   INTEGER   required)
  *
- * IN:  meshID
+ * IN:  mesh, meshNameLength
  * OUT: required(1:true, 0:false)
  *
  * @copydoc precice::SolverInterface::requiresMeshConnectivityFor()
  */
 PRECICE_API void precicef_requires_mesh_connectivity_for_(
-    const int *meshID,
-    int *      required);
+    const char *meshName,
+    int *       required,
+    int         meshNameLength);
 
 /**
  * Fortran syntax:
  * precicef_set_vertex(
- *   INTEGER          meshID,
+ *   CHARACTER        meshName(*),
  *   DOUBLE PRECISION position(dim),
  *   INTEGER          vertexID )
  *
- * IN:  meshID, position
+ * IN:  mesh, position, meshNameLength
  * OUT: vertexID
  *
  * @copydoc precice::SolverInterface::setMeshVertex()
  *
  */
 PRECICE_API void precicef_set_vertex_(
-    const int *   meshID,
+    const char *  meshName,
     const double *position,
-    int *         vertexID);
+    int *         vertexID,
+    int           meshNameLength);
```

Comment:
```suggestion
    int           meshNameLength),
    int *         vertexID;
```
same logic as above.

Reply:
same reasoning